pkg/syncer/metadatasyncer.go

```diff
@@ -1059,6 +1059,41 @@ func csiUpdatePod(ctx context.Context, pod *v1.Pod, metadataSyncer *metadataSync
 					},
 				}
 
+				log.Debugf("Calling UpdateVolumeMetadata for volume %s with updateSpec: %+v", updateSpec.VolumeId.Id, spew.Sdump(updateSpec))
+				if err := metadataSyncer.volumeManager.UpdateVolumeMetadata(ctx, updateSpec); err != nil {
+					log.Errorf("UpdateVolumeMetadata failed for volume %s with err: %v", volume.Name, err)
+				}
+			}
+		} else {
+			if volume.VsphereVolume != nil && metadataSyncer.coCommonInterface.IsFSSEnabled(ctx, common.CSIMigration) {
+				var metadataList []cnstypes.BaseCnsEntityMetadata
+				var podMetadata *cnstypes.CnsKubernetesEntityMetadata
+				if !deleteFlag {
+					podMetadata = cnsvsphere.GetCnsKubernetesEntityMetaData(pod.Name, nil, deleteFlag, string(cnstypes.CnsKubernetesEntityTypePOD), pod.Namespace, metadataSyncer.configInfo.Cfg.Global.ClusterID, nil)
+				} else {
+					podMetadata = cnsvsphere.GetCnsKubernetesEntityMetaData(pod.Name, nil, deleteFlag, string(cnstypes.CnsKubernetesEntityTypePOD), pod.Namespace, metadataSyncer.configInfo.Cfg.Global.ClusterID, nil)
+				}
+				metadataList = append(metadataList, cnstypes.BaseCnsEntityMetadata(podMetadata))
+				containerCluster := cnsvsphere.GetContainerCluster(metadataSyncer.configInfo.Cfg.Global.ClusterID, metadataSyncer.configInfo.Cfg.VirtualCenter[metadataSyncer.host].User, metadataSyncer.clusterFlavor)
+				var volumeHandle string
+				var err error
+				migrationVolumeSpec := migration.VolumeSpec{VolumePath: volume.VsphereVolume.VolumePath}
+				volumeHandle, err = volumeMigrationService.GetVolumeID(ctx, migrationVolumeSpec)
```

Comment:
what if volumeMigrationService is nil? the case you observed in the issue - https://github.com/kubernetes-sigs/vsphere-csi-driver/issues/360

Reply:
Yes, so the other PR https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/364 is taking care of this here:
https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/364/files#diff-6fc246216efb3eb9047f82d1116e9813R1060